tests/test_verify_receipt.py

```diff
@@ -0,0 +1,428 @@
+"""
+Positive test cases for verify_receipt
+"""
+
+import os
+import json
+import io
+import unittest
+import shutil
+import tempfile
+from contextlib import redirect_stdout
+
+from datatrails_scitt_samples.datatrails.servicecontext import ServiceContext
+
+from datatrails_scitt_samples.datatrails.eventpreimage import get_event
+from datatrails_scitt_samples.datatrails.entryid import entryid_to_identity
+
+
+from datatrails_scitt_samples.scripts.generate_example_key import (
+    main as generate_example_key,
+)
+from datatrails_scitt_samples.scripts.create_hashed_signed_statement import (
+    main as create_hashed_signed_statement,
+)
+from datatrails_scitt_samples.scripts.register_signed_statement import (
+    main as register_signed_statement,
+)
+from datatrails_scitt_samples.scripts.verify_receipt import (
+    main as verify_receipt,
+)
+
+
+class TestVerifyReciept(unittest.TestCase):
+    """
+    Tests verification of a known receipt.
+    """
+
+    def setUp(self):
+        self.test_dir = tempfile.mkdtemp()
+        self.parent_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+        self.bad_json_file = os.path.join(self.test_dir, "bad.json")
+        with open(self.bad_json_file, "w") as file:
+            file.write("this is not json")
+
+    def tearDown(self):
+        shutil.rmtree(self.test_dir)
+
+    @unittest.skipUnless(
+        os.getenv("DATATRAILS_CLIENT_SECRET") != "",
+        "test requires authentication via env DATATRAILS_xxx",
+    )
+    def test_verify_failed_for_tampered_event(self):
+        """
+        registers a statement then verifies its receipt
+        """
+        # generate an example key
+        generate_example_key(["--signing-key-file", "my-signing-key.pem"])
+
+        # create a signed statement
+        create_hashed_signed_statement(
+            [
+                "--signing-key-file",
+                "my-signing-key.pem",
+                "--payload-file",
+                os.path.join(
+                    self.parent_dir,
+                    "datatrails_scitt_samples",
+                    "artifacts",
+                    "thedroid.json",
+                ),
+                "--content-type",
+                "application/json",
+                "--subject",
+                "TestRegisterSignedStatement:test_create_and_register_statement",
+                "--issuer",
+                "https://github.com/datatrails/datatrails-scitt-samples",
+                "--output-file",
+                f"{self.test_dir}/signed-statement.cbor",
+            ]
+        )
+        self.assertTrue(os.path.exists(f"{self.test_dir}/signed-statement.cbor"))
+
+        # register the signed statement
+        output = io.StringIO()
+        with redirect_stdout(output):
+            register_signed_statement(
+                [
+                    "--signed-statement-file",
+                    f"{self.test_dir}/signed-statement.cbor",
```

Comment:
if composing filepaths use os.path.join

Reply:
I really don't think that matters on any platform we support (including windows) but ok